src/DIRAC/WorkloadManagementSystem/Client/PilotLoggingPlugins/FileCacheLoggingPlugin.py

```diff
@@ -0,0 +1,133 @@
+"""
+File cache logging plugin.
+"""
+import os
+import json
+import re
+from DIRAC import S_OK, S_ERROR, gLogger
+
+sLog = gLogger.getSubLogger(__name__)
+
+
+class FileCacheLoggingPlugin:
+    """
+    File cache logging. Log records are appended to a file, one for each pilot.
+    It is assumed that an agent will be installed together with this plugin, which will copy
+    the files to a safe place and clear the cache.
+    """
+
+    def __init__(self):
+        """
+        Sets the pilot log files location for a WebServer.
+
+        """
+        # UUID pattern
+        self.pattern = re.compile(r"^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+        # pilot stamp pattern
+        self.stamppattern = re.compile(r"^[0-9A-F]{8}$")
+        self.meta = {}
+        logPath = os.path.join(os.getcwd(), "pilotlogs")
+        self.meta["LogPath"] = logPath
+        if not os.path.exists(logPath):
+            os.makedirs(logPath)
+        sLog.info("Pilot logging directory:", logPath)
+
+    def sendMessage(self, message, pilotUUID, vo):
+        """
+        File cache sendMessage method. Write the log message to a file line by line.
+
+
+        :param message: text to log in json format
+        :type message: str
```

Comment:
Could be

```suggestion
        :param str message: text to log in json format
```

(applies to other parameters too)